src/configure.ac

```diff
@@ -379,6 +389,24 @@ if test "x${with_hdf5}" != "xno"; then
 	fi
 fi
 
+if test "x${with_adios}" != "xno"; then
+	adios2_config="${ADIOS_DIR}/bin/adios2-config"
```

Comment:
I don't see where `ADIOS2_USE_MPI` is set to true.

It has to be:
1. Add `ADIOS2_USE_MPI=1` in the `if` in `configure.ac`
2. Add `AC_SUBST(ADIOS2_USE_MPI)` at the bottom of `configure.ac`
3. Add `#undef ADIOS2_USE_MPI` to in `src/config.h.in`

Reply:
ADIOS2_USE_MPI is an ADIOS config flag (the adios2-config returns it in --flags string) - I'm abusing it a little now

This is to be changed I think:
-  Option 1: If TCLB is MPI then ADIOS must be MPI and vice versa, nice but it doesn't make sense even for testing imho?
- Option 2:  Enable adios in MPI version of TCLB and require ADIOS2_USE_MPI.

Reply:
I think we need our own flag for enabling ADIOS then. Remember that if an flag is in the compilation (like `-DFLAG`) and not in the source (like `#define FLAG 1`) then `make` cannot see the change and will not recompile properly. Please make a flag for the ADIOS support as described above (maybe `WITH_ADIOS`)

TCLB is always MPI, we don't have an option for that.